From 3ee8bd778135ca3b2da9b5fd30d3ef8d2a93b99c Mon Sep 17 00:00:00 2001
From: Ningxin Hu <ningxin@intel.com>
Date: Wed, 31 Aug 2016 14:18:55 +0000
Subject: [PATCH 1/3] [ostro-os-xt] Fallback to MADV_DONTNEED

Fallback to MADV_DONTNEED, as MADV_FREE causes madvise always
returns -1 on Ostro OS XT.
---
 third_party/WebKit/Source/wtf/PageAllocator.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/third_party/WebKit/Source/wtf/PageAllocator.cpp b/third_party/WebKit/Source/wtf/PageAllocator.cpp
index 7f7a884..03d5a64 100644
--- a/third_party/WebKit/Source/wtf/PageAllocator.cpp
+++ b/third_party/WebKit/Source/wtf/PageAllocator.cpp
@@ -224,7 +224,9 @@ void decommitSystemPages(void* addr, size_t len)
 {
     ASSERT(!(len & kSystemPageOffsetMask));
 #if OS(POSIX)
-    int ret = madvise(addr, len, MADV_FREE);
+    // Fallback to MADV_DONTNEED, as MADV_FREE causes
+    // madvise always returns -1 on Ostro OS XT.
+    int ret = madvise(addr, len, MADV_DONTNEED);
     RELEASE_ASSERT(!ret);
 #else
     setSystemPagesInaccessible(addr, len);
-- 
2.8.4

